### PR TITLE
Commonify video_core utility headers

### PR DESCRIPTION
**Pull request by archshift**
These two headers (`color.h` and `math.h`) both are very useful outside of video_core, and don't assume anything inherent about the 3DS or any of its internals. Therefore, they should be in common instead. 

I renamed `math.h` to `vector_math.h` because it only contained code for doing vector math anyway.

### CODE REVIEW

File: src/video_core/debug_utils/debug_utils.h [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include <mutex>
 #include <vector>
 
-#include "video_core/math.h"
+#include "common/vector_math.h"
```
Comment (linkmauve) [inline src/video_core/debug_utils/debug_utils.h]: Add a blank line here, as it’s done elsewhere to separate common from video_core.

**Comment by linkmauve** [comment]
LGTM.

File: src/video_core/debug_utils/debug_utils.h (hunk 1 above)
Comment (archshift) [inline src/video_core/debug_utils/debug_utils.h]: done

File: src/video_core/pica.h [hunk 2]
```diff
@@ -14,10 +14,9 @@
 #include "common/bit_field.h"
 #include "common/common_funcs.h"
 #include "common/common_types.h"
+#include "common/vector_math.h"
```
Comment (linkmauve) [inline src/video_core/pica.h]: Wrong order of include, vector comes after logging.

File: src/citra_qt/debugger/graphics_cmdlists.cpp [hunk 3]
```diff
@@ -11,9 +11,9 @@
 #include <QSpinBox>
 #include <QComboBox>
 
-#include "video_core/pica.h"
-#include "video_core/math.h"
+#include "common/vector_math.h"
 
+#include "video_core/pica.h"
 #include "video_core/debug_utils/debug_utils.h"
```
Comment (yuriks) [inline src/citra_qt/debugger/graphics_cmdlists.cpp]: These two lines aren't sorted.

File: src/video_core/pica.h (hunk 2 above)
Comment (yuriks) [inline src/video_core/pica.h]: +1

**Comment by yuriks** [comment]
LGTM

**Comment by neobrain** [comment]
LGTM

**Pull request merged**